kontakt-reservasjonsregister-client/src/main/java/no/difi/meldingsutveksling/serviceregistry/krr/DSFClient.java

```diff
@@ -1,18 +1,26 @@
 package no.difi.meldingsutveksling.serviceregistry.krr;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nimbusds.jose.proc.BadJWSException;
+import no.difi.move.common.oauth.JWTDecoder;
+import no.difi.move.common.oauth.KeystoreHelper;
 import org.springframework.http.*;
 import org.springframework.web.client.RestTemplate;
 
+import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.security.cert.CertificateException;
 
 public class DSFClient {
 
     private URL endpointURL;
+    private KeystoreHelper keystoreHelper;
 
-    public DSFClient(URL endpointURL) {
+    public DSFClient(URL endpointURL, KeystoreHelper keystoreHelper) {
         this.endpointURL= endpointURL;
+        this.keystoreHelper = keystoreHelper;
     }
 
     public DSFResource getDSFResource(String identifier, String token) throws KRRClientException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getDSFResource" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)